dask/dataframe/groupby.py

```diff
@@ -2119,10 +2122,11 @@ def nunique(self, split_every=None, split_out=1):
     @derived_from(pd.core.groupby.SeriesGroupBy)
     def aggregate(self, arg, split_every=None, split_out=1):
         result = super().aggregate(arg, split_every=split_every, split_out=split_out)
-        if self._slice:
+        if self._slice is not None:
             result = result[self._slice]
 
         if not isinstance(arg, (list, dict)) and isinstance(result, DataFrame):
+            assert len(result.columns) == 1
             result = result[result.columns[0]]
```

Comment:
This bit is extra confusing - before this PR, the output would depend on the column order:
```python
[ins] In [1]: import dask.dataframe as dd
         ...: import pandas as pd

[ins] In [2]: ddf = dd.from_pandas(pd.DataFrame([[0, 1, 2]], columns=[0, 1, 2]), npartitions=1)

[ins] In [3]: ddf.groupby(2)[0].agg("sum").name
Out[3]: 0

[ins] In [4]: ddf.columns = [2, 1, 0]

[ins] In [5]: ddf.groupby(2)[0].agg("sum").name
Out[5]: 1
```

Effectively, the `if self._slice:` would be `False`, but then `if not isinstance(arg, (list, dict)) and isinstance(result, DataFrame)` would be True, and then we'd **filter to the first column**.

I'm not sure under what condition we hit `if not isinstance(arg, ...` when using a single agg func (as we are inside `SeriesGroupBy`), but I think the assumption is that there'd only be 1 column in that DataFrame, so I added the assert. If that's assumption is right, is an `assert` fine (I see a couple others in this file) or should this be a `raise`d exception?